Tests/CodegenCLITests/VersionCheckerTests.swift

```diff
@@ -35,216 +35,268 @@ class VersionCheckerTests: XCTestCase {
     // given
-    try fileManager.createFile(
-      body: version2PackageResolvedFileBody(apolloVersion: Constants.CLIVersion),
-      named: "Package.resolved",
-      inDirectory: "MyProject.xcodeproj/project.xcworkspace/xcshareddata/swiftpm"
-    )
-
-    // when
-    let result = try VersionChecker.matchCLIVersionToApolloVersion(
-      projectRootURL: fileManager.directoryURL
-    )
-
-    // then
-    expect(result).to(equal(.versionMatch))
+    for packageVersion in packageResolvedVersion.allCases {
+      try testPackageResolvedFile(
+        packageVersion: packageVersion,
+        inDirectory: "MyProject.xcodeproj/project.xcworkspace/xcshareddata/swiftpm",
+        apolloVersion: Constants.CLIVersion
+      ) {
+
+        // when
+        let result = try VersionChecker.matchCLIVersionToApolloVersion(
+          projectRootURL: fileManager.directoryURL
+        )
+
+        // then
+        expect(result).to(equal(.versionMatch))
+      }
+    }
   }
 
-  func test__matchCLIVersionToApolloVersion__givenPackageResolvedFileInXcodeProject_withVersion2FileFormat_hasNonMatchingVersion_returns_versionMismatch() throws {
+  func test__matchCLIVersionToApolloVersion__givenPackageResolvedFileInXcodeProject_withKnownResolvedFileFormats_hasNonMatchingVersion_returns_versionMismatch() throws {
     // given
     let apolloVersion = "1.0.0.test-1"
-    try fileManager.createFile(
-      body: version2PackageResolvedFileBody(apolloVersion: apolloVersion),
-      named: "Package.resolved",
-      inDirectory: "MyProject.xcodeproj/project.xcworkspace/xcshareddata/swiftpm"
-    )
 
-    // when
-    let result = try VersionChecker.matchCLIVersionToApolloVersion(
-      projectRootURL: fileManager.directoryURL
-    )
-
-    // then
-    expect(result).to(equal(.versionMismatch(cliVersion: Constants.CLIVersion, apolloVersion: apolloVersion)))
+    for packageVersion in packageResolvedVersion.allCases {
+      try testPackageResolvedFile(
+        packageVersion: packageVersion,
+        inDirectory: "MyProject.xcodeproj/project.xcworkspace/xcshareddata/swiftpm",
+        apolloVersion: apolloVersion
+      ) {
+
+        // when
+        let result = try VersionChecker.matchCLIVersionToApolloVersion(
+          projectRootURL: fileManager.directoryURL
+        )
+
+        // then
+        expect(result).to(equal(
+          .versionMismatch(cliVersion: Constants.CLIVersion, apolloVersion: apolloVersion)
+        ))
+      }
+    }
   }
   
-  func test__matchCLIVersionToApolloVersion__givenPackageResolvedFileInXcodeWorkspaceAndProject_withVersion2FileFormat_hasMatchingVersion_returns_versionMatch_fromWorkspace() throws {
+  func test__matchCLIVersionToApolloVersion__givenPackageResolvedFileInXcodeWorkspaceAndProject_withKnownResolvedFileFormats_hasMatchingVersion_returns_versionMatch_fromWorkspace() throws {
```

Comment:
This test was changed to have both the workspace and project use the same Apollo version. Previously they were different and that could also affect the outcome. So this test was changed and there is a new test below that has mismatched versions to ensure that the workspace one is still selected.